### PR TITLE
refactor(NgClass): remove pipes from property bindings

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
This is the refactoring needed to support #2025

**Comment by pkozlowski-opensource** [comment]
@vsavkin this is the refactoring to the `NgClass` directive we've discussed yesterday - it essentially removes pipes from the `@Directives({properties: [...]})`. Would be awesome if you could review this one so I know if I'm on the right track....

File: modules/angular2/src/directives/class.ts [hunk 1]
```diff
@@ -17,12 +22,20 @@ export class CSSClass {
     }
   }
 
-  set iterableChanges(changes) {
-    if (isPresent(changes)) {
-      changes.forEachAddedItem((record) => { this._toggleClass(record.key, record.currentValue); });
-      changes.forEachChangedItem(
+  onCheck() {
+    if (this.rawClass != this._prevRawClass) {
```
Comment (vsavkin) [inline modules/angular2/src/directives/class.ts]: You can do this in the rawClass setter.

**Comment by vsavkin** [comment]
Merged via https://github.com/angular/angular/commit/ad7aca631d6cb7a2029db0a0d44c3cd728567d85

**Pull request closed without merging**